lib/sinon/util/fake_xdomain_request.js

```diff
@@ -7,8 +7,24 @@
 /**
  * Fake XDomainRequest object
  */
+
+/**
+ * Returns the global to prevent assigning values to 'this' when this is undefined.
+ * This can occur when files are interpreted by node in strict mode.
+ * @private
+ */
+function getGlobal() {
+    "use strict";
+
+    return typeof window !== "undefined" ? window : global;
```

Comment:
If you modify this function to use `self` instead of `window`, then the code will not fail in Web Workers (where the `window` object is absent).

```javascript
return self || global;
```

Whether the rest of the file will work, I don't know.